examples/_demo_for_PR.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+import ase
+from equistore import TensorBlock, TensorMap, Labels
+from equisolve.vanilla.preprocessing import ase_frames_to_properties_tensormap
+
+# artificial data
+
+n_strucs = 2
+energies = np.random.rand(n_strucs)
+forces = np.random.rand(n_strucs, 5, 3)
+
+frames = []
+for i in range(len(energies)):
+    frame = ase.Atoms('CH4')
+    frame.info['energy'] = energies[i]
+    frame.arrays['forces'] = forces[i]
+    frames.append( frame )
+
+get_energy = lambda frame : frame.info['energy']
```

Comment:
Nice idea with the extraction functions. If we use this the user can even only provide the string of the `energy` and forces. Might be worth what @Luthaf  and @rosecers think of this design.

Reply:
I think that we should also have a function that takes in string as input. Something like 

```py
energy = properties_to_tensormap(frames, property="energy", positions_gradient="forces_dft", cell_gradient="virial_dft")
```

that can also take lambda for more flexibilty:
```py
energy = properties_to_tensormap(
    frames, 
    property=lambda frame: frame.info["energy"] * RY_TO_EV, 
    positions_gradient=lambda frame: -frame.array["forces"] * RY_AU_TO_EV_A,
    cell_gradient=lambda frame: frame.array["virial"] * 3.4,
)
```

The string version can then dispatch depending on the frame type (for ase it would use `frame.info` and `frame.arrays`, for chemfile something else ...)

---

The main issue I see with this is how to deal with the fact that we want gradient w.r.t. positions, but forces are **minus** that. I'm not sure how we can both have a general "extract properties from frames" and handle forces transparently-ish to the user.

Reply:
> The main issue I see with this is how to deal with the fact that we want gradient w.r.t. positions, but forces are minus that.

Proposition 1:
We call the gradient parameter "negative_positions" and not "positions"? A bit clumsy, but the user will stumble on this, read the doc, understand and continue with his life.

Proposition 2:
We just keep "positions" and work with gradients, because the user who is not invested in the library will only extract the forces out of the object through some utility in this library which knows transforms gradients to forces. So even if the user does not know that we store gradients in the object, he/she will not have problems with this.

I am for proposition 2

Reply:
I see the user has to think about it when the default way that extracts the forces out of the frame does not work. 

But can't we then just offer two interfaces
```python
def frames_to_energy_tensormap(frames, energy=None, forces=None, stress=None):
    if isinstance(forces, function):
        forces = lambda frame : -forces(frame)
    frames_to_property_tensormap(frames, energy, forces, stress)

def frames_to_property_tensormap(frames, property=None, positions_gradient=None, cell_gradient=None, is_structure_property=True):
....
Naming is up for debate

Reply:
see newest commit, it is taking this approach with different naming